src/store/index.js

```diff
@@ -1,25 +1,60 @@
 /**
- * The data store is currently a JSON file, but its schema mirrors the Gmail
- * API so we can feed it from your own inbox at a later date.
- *
  * @flow
  */
 
 import type { Data, Message } from './types';
 
 const store: Data = require('./emails.json');
 
-/**
- * You might find it useful to implement some functions that make accessing
- * deeply nested attributes of Messages or Threads...
- */
+function formatTimestamp(timestamp) {
+  const now = Date.now();
+  const midnightToday = new Date().setUTCHours(0, 0, 0, 0);
+  const timestampMilliseconds = Date.parse(timestamp);
+  const justDate = new Date(timestamp).toLocaleDateString();
+  const justTime = new Date(timestamp).toLocaleTimeString([], {
+    hour: '2-digit',
+    minute: '2-digit',
+  });
+  if (timestampMilliseconds < now && timestampMilliseconds > midnightToday) {
+    return justTime;
+  }
+  return justDate;
+}
+
+function formatSender(sender) {
+  return sender.match(/.+?(?=\\u003)/)[0];
+}
+
+function getHeaderValue(headers, name) {
+  const { value } = headers.filter(header => header.name === name)[0];
+  if (name === 'Date') {
+    return formatTimestamp(value);
+  }
+  if (name === 'From') {
+    return formatSender(value);
+  }
+  return value;
+}
 
-// For example, this returns the subject of a given message if it is defined
-// by a header in the message payload
-export function getMessageSubject(message: Message): ?string {
-  const { headers } = message.payload;
-  const subjectHeader = headers.find(header => header.name === 'Subject');
-  return subjectHeader && subjectHeader.value;
+export function getThreads() {
+  const { mailboxes, threads } = store;
+  const inbox = mailboxes.INBOX;
+  const threadObjects = inbox.threadIds.map((id) => {
+    const [newestMessage] = threads[id].messages.slice(-1);
+    const message = store.messages[newestMessage.id];
+    const { headers } = message.payload;
+    const sender = getHeaderValue(headers, 'From');
```

Comment:
I think if you remove these four named variables and just return the formatted versions of what you want in your object below, it will improve the readability here. Like this:

```js
return {
  sender: getHeaderValue(headers, 'From'),
  timestamp: getHeaderValue(headers, 'Date'),
  subject: getHeaderValue(headers, 'Subject') || '(No Subject)';,
  snippet: message.snippet,
};
```